profiler/src/ProfilerEngine/Datadog.Profiler.Native/CollectorBase.h

```diff
@@ -112,9 +110,9 @@ class CollectorBase
         return input;
     }
 
-    std::list<Sample> TransformRawSamples(const std::list<TRawSample>& input)
+    std::list<std::shared_ptr<Sample>> TransformRawSamples(std::list<TRawSample>&& input)
```

Comment:
Why not keeping passing a const ref list?

Reply:
mainly semantic reason:
In `GetSamples`, we create the list, pass by `const &` the list to `TransformRawSamples` and then the list is not used.
Here it's just to say `I transfer the ownership of the list to TransformRawSamples because I do not need it anymore`